### PR TITLE
Implemented Routing

### PR DESCRIPTION
**Pull request by Alee-star**
Implemented routing for all navigation items.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,13 +1,25 @@
...
+import Home from "./components/Home";
+import Leaderboard from "./components/Leaderboard";
+import Challenges from "./components/Challenges";
 import "./App.css";
 
 function App() {
   return (
     <div className="App">
       <Heading />
       <GalleryList />
```
Comment (vonnue-nidheesh) [inline src/App.js]: Remove the galleryList from here. It should be inside the videos/photos page

File: src/App.js [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+      <Routes>
+        <Route path="/" element={<Navigate to="/Home" />} />
+        <Route path="/home" element={<Home />} />
```
Comment (vonnue-nidheesh) [inline src/App.js]: `<Route path="/" element={<Home />} />`

For now, there is no need for a specific route for home.

File: src/App.js [hunk 3]
```diff
@@ -1,13 +1,25 @@
 import React from "react";
-import Heading from "./components/Header/Heading";
+import { Route, Routes, Navigate } from "react-router-dom";
```
Comment (vonnue-nidheesh) [inline src/App.js:2]: pages must be under a directory named pages.

Comment (vonnue-nidheesh) [reply src/App.js:2]: eg:

```
pages
- home
- videos
- leaderboard

components
- galleryList
- navbar
```

File: src/components/Favourites/Favourites.jsx [hunk 4]
```diff
@@ -2,8 +2,8 @@ import React from "react";
 import ImageCard from "../Banner/ImageCard";
 import "./Favourites.css";
 
-const Favourites = ({ photos, addToFavourites }) => {
-  if (!photos.length) {
+const Favourites = ({ photos = [], addToFavourites }) => {
+  if (!photos || photos.length === 0) {
```
Comment (vonnue-nidheesh) [inline src/components/Favourites/Favourites.jsx]: `if (!photos?.length) {`

File: src/components/Home.jsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import React from "react";
+
+const Home = () => {
+  return;
+};
+
+export default Home;
```
Comment (vonnue-nidheesh) [inline src/components/Home.jsx]: No need of this component.

File: src/index.js [hunk 6]
```diff
@@ -1,14 +1,18 @@
...
+import reportWebVitals from "./reportWebVitals";
+import { BrowserRouter } from "react-router-dom";
 
-const root = ReactDOM.createRoot(document.getElementById('root'));
+const root = ReactDOM.createRoot(document.getElementById("root"));
 root.render(
-  <React.StrictMode>
+  // <React.StrictMode>
+  //   <App />
+  // </React.StrictMode>
```
Comment (vonnue-nidheesh) [inline src/index.js]: Remove this commented code or comment something like 'Note: this is the previous version before implementing router'.

File: src/App.js (hunk 2 above)
Comment (Alee-star) [reply src/App.js]: <NavLink
                to={`/${item.toLowerCase()}`}
                className={({ isActive }) => (isActive ? "active" : "")}
                onClick={() => handleTabChange(item)}
              >
                {item}
              </NavLink> path is given like this. so home is necassary

**Review by vonnue-nidheesh — APPROVED** [review]

**Pull request merged**